### ISSUE TITLE
Impossibility to scan a directory in a public storage account

### ISSUE THREAD

**Issue by NicolasDorier** [Azure/azure-storage-azcopy#971]
My account is public.
```
./azcopy cp "https://$AZURE_ACCOUNT_NAME.blob.core.windows.net/$AZURE_STORAGE_CONTAINER/$DIRECTORY_NAME/*" "dist"
INFO: Scanning...

failed to perform copy command due to error: no SAS token or OAuth token is present and the resource is not public
```
Version `azcopy version 10.4.0`.

It worked long ago... unsure when it got broken.

## How to reproduce

This works:
https://btcpayservertravis.blob.core.windows.net/dist?prefix=dist-618031009&restype=container&comp=list

This does not work:
```
./azcopy cp "https://btcpayservertravis.blob.core.windows.net/dist/dist-618031009/*" "dist"
```

**Comment by JohnRusk** [comment]
Thanks for logging this.

**Comment by nakulkar-msft** [comment]
Fixed in 10.6.1
